[Feature] Compatibility with Atlas API v3

**Is your feature request related to a problem? Please describe.**

Current `main` fails a number of tests when in an environment with [latest atlas API (v3, currently in dev)](https://github.com/brainglobe/brainglobe-atlasapi/pull/769). Logging representative local `pytest` failures here, to help diagnosing and future PR review.

These are just the symptoms for now.

Sub-issue of https://github.com/brainglobe/brainglobe-atlasapi/issues/780

### Underlying file structure has changed

These are very much expected

```bash
 FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/gw/g898mw1d2319ncb69pn6gvjc0000gn/T/pytest-of-alessandrofelder/pytest-3/test_config_edit0/new_brainglobe_dir/allen_mouse_100um_v3_0/structures.json'

FAILED tests/test_unit/test_metadata_reading.py::test_metadata_reading - FileNotFoundError: [Errno 2] No such file or directory: '/private/var/folders/gw/g898mw1d2319ncb69pn6gvjc0000gn/T/pytest-of-alessandrofelder/pytest-3/test_config_edit0/new_brainglobe_dir/example_mouse_100um_v3_0/metadata.json'

where exists = PosixPath('/Users/alessandrofelder/.brainglobe-tests/.brainglobe/example_mouse_100um_v1.2').exists
```

### Some breakages in some functions

```bash
assert selected_atlas_name in get_downloaded_atlases()

  tests/test_unit/test_napari_atlas_representation.py::test_add_additional_reference - TypeError: 'NoneType' object is not subscriptable

  tests/test_unit/test_napari_atlas_representation.py::test_add_to_viewer[osten_mouse_100um-True] - AssertionError: assert False
 +  where False = allclose(array([[  0.,   0.,   0.],\n       [ 43.,  39., 113.]]), array([[  0.,   0.,   0.],\n       [131.,  79., 113.]]))
 +    where array([[  0.,   0.,   0.],\n       [ 43.,  39., 113.]]) = Extent(data=array([[  0.,   0.,   0.],\n       [ 43.,  39., 113.]]), world=array([[  0.,   0.,   0.],\n       [ 43.,  39., 113.]]), step=array([1., 1., 1.])).world
 +      where Extent(data=array([[  0.,   0.,   0.],\n       [ 43.,  39., 113.]]), world=array([[  0.,   0.,   0.],\n       [ 43.,  39., 113.]]), step=array([1., 1., 1.])) = <Labels layer 'osten_mouse_100um_annotation' at 0x31c49ac10>.extent
 +    and   array([[  0.,   0.,   0.],\n       [131.,  79., 113.]]) = Extent(data=array([[  0.,   0.,   0.],\n       [131.,  79., 113.]]), world=array([[  0.,   0.,   0.],\n       [131.,  79., 113.]]), step=array([1., 1., 1.])).world
 +      where Extent(data=array([[  0.,   0.,   0.],\n       [131.,  79., 113.]]), world=array([[  0.,   0.,   0.],\n       [131.,  79., 113.]]), step=array([1., 1., 1.])) = <Image layer 'osten_mouse_100um_reference' at 0x3073dfed0>.extent
```

### Some Qt Signals are never emitted, some functions are not called

```
 FAILED tests/test_integration/test_brainrender_viewer_widget.py::test_structure_row_double_clicked - AssertionError: Expected 'add_structure_to_viewer' to be called once. Called 0 times.
FAILED tests/test_integration/test_brainrender_viewer_widget.py::test_add_additional_reference_selected - AssertionError
FAILED tests/test_integration/test_brainrender_viewer_widget.py::test_show_structures_checkbox - AssertionError: expected call not found.
Expected: refresh('example_mouse_100um', False)
  Actual: not called.
FAILED tests/test_unit/test_atlas_viewer_view.py::test_double_click_on_locally_available_atlas_row[0-example_mouse_100um] - pytestqt.exceptions.TimeoutError: Signal add_atlas_requested(QString) not emitted after 5000 ms
FAILED tests/test_unit/test_atlas_viewer_view.py::test_double_click_on_locally_available_atlas_row[4-allen_mouse_100um] - pytestqt.exceptions.TimeoutError: Signal add_atlas_requested(QString) not emitted after 5000 ms
FAILED tests/test_unit/test_atlas_viewer_view.py::test_double_click_on_locally_available_atlas_row[14-osten_mouse_100um] - pytestqt.exceptions.TimeoutError: Signal add_atlas_requested(QString) not emitted after 5000 ms
FAILED tests/test_unit/test_atlas_viewer_view.py::test_additional_reference_menu - pytestqt.exceptions.TimeoutError: Signal additional_reference_requested(QString) not emitted after 5000 ms
```

**Describe the solution you'd like**

This issue is purely to ensure the current behaviour persists when we switch to Atlas API v3.
So this issue can be closed when we convince ourselves with `pytest` and manual user testing that there is no change in user-facing functionality between current Atlas API and new Atlas API.

We may also want to rethink the functionality we provide (e.g. #241) but that can be tracked separately.

--- a/brainrender_napari/data_models/atlas_table_model.py
+++ b/brainrender_napari/data_models/atlas_table_model.py
@@ -9,6 +9,10 @@ from qtpy.QtGui import QBrush, QColor
 from qtpy.QtWidgets import QTableView
 
 from brainrender_napari.utils.formatting import format_atlas_name
+from brainrender_napari.utils.load_user_data import (
+    read_atlas_metadata_from_file,
+    read_atlas_structures_from_file,
+)
 
 
 class AtlasTableModel(QAbstractTableModel):
@@ -28,6 +32,8 @@ class AtlasTableModel(QAbstractTableModel):
         )
         self.view_type: QTableView = view_type
         self.refresh_data()
+        read_atlas_metadata_from_file.cache_clear()
+        read_atlas_structures_from_file.cache_clear()
 
     def refresh_data(self):
         """Refresh model data by calling atlas API"""

--- a/brainrender_napari/data_models/atlas_table_model.py
+++ b/brainrender_napari/data_models/atlas_table_model.py
@@ -41,9 +41,7 @@ class AtlasTableModel(QAbstractTableModel):
                     [
                         name,
                         format_atlas_name(name),
-                        folder_version_to_dotted(
-                            get_local_atlas_version(name)
-                        ),
+                        get_local_atlas_version(name),
                         latest_version,
                     ]
                 )

--- a/brainrender_napari/data_models/atlas_table_model.py
+++ b/brainrender_napari/data_models/atlas_table_model.py
@@ -1,5 +1,4 @@
 from brainglobe_atlasapi.list_atlases import (
-    folder_version_to_dotted,
     get_all_atlases_lastversions,
     get_atlases_lastversions,
     get_local_atlas_version,

--- a/brainrender_napari/data_models/atlas_table_model.py
+++ b/brainrender_napari/data_models/atlas_table_model.py
@@ -1,4 +1,5 @@
 from brainglobe_atlasapi.list_atlases import (
+    folder_version_to_dotted,
     get_all_atlases_lastversions,
     get_atlases_lastversions,
     get_local_atlas_version,
@@ -40,7 +41,9 @@ class AtlasTableModel(QAbstractTableModel):
                     [
                         name,
                         format_atlas_name(name),
-                        get_local_atlas_version(name),
+                        folder_version_to_dotted(
+                            get_local_atlas_version(name)
+                        ),
                         latest_version,
                     ]
                 )

--- a/brainrender_napari/data_models/atlas_table_model.py
+++ b/brainrender_napari/data_models/atlas_table_model.py
@@ -32,11 +32,13 @@ class AtlasTableModel(QAbstractTableModel):
         )
         self.view_type: QTableView = view_type
         self.refresh_data()
-        read_atlas_metadata_from_file.cache_clear()
-        read_atlas_structures_from_file.cache_clear()
 
     def refresh_data(self):
         """Refresh model data by calling atlas API"""
+        # atlases may have been downloaded or updated since the last refresh
+        read_atlas_metadata_from_file.cache_clear()
+        read_atlas_structures_from_file.cache_clear()
+
         all_atlases: dict[str, str] = get_all_atlases_lastversions()
         local_atlases = get_atlases_lastversions().keys()
         data = []

--- a/brainrender_napari/napari_atlas_representation.py
+++ b/brainrender_napari/napari_atlas_representation.py
@@ -42,7 +42,7 @@ class NapariAtlasRepresentation:
         The layers are connected to the mouse move callback to setup tooltip.
         """
         reference = self.viewer.add_image(
-            self.bg_atlas.reference,
+            self.bg_atlas.template,
             name=f"{self.bg_atlas.atlas_name}_reference",
         )
 

--- a/brainrender_napari/utils/load_user_data.py
+++ b/brainrender_napari/utils/load_user_data.py
@@ -6,7 +6,18 @@ from brainglobe_atlasapi import BrainGlobeAtlas
 @lru_cache(maxsize=32)
 def read_atlas_metadata_from_file(atlas_name: str):
     """Reads atlas metadata from the local manifest in the BrainGlobe
-    directory.
+    directory. Removes nested keys that are not relevant for the user.
+    Converts the additional_references list of dicts to a list of names.
+
+    Parameters
+    ----------
+    atlas_name : str
+        Name of the atlas to read metadata for.
+
+    Returns
+    -------
+    dict
+        A dictionary containing the atlas metadata.
     """
     atlas = BrainGlobeAtlas(atlas_name=atlas_name, check_latest=False)
 
@@ -33,6 +44,16 @@ def read_atlas_metadata_from_file(atlas_name: str):
 def read_atlas_structures_from_file(atlas_name: str):
     """Reads atlas structure info from the local terminology file in the
     BrainGlobe directory.
+
+    Parameters
+    ----------
+    atlas_name : str
+        Name of the atlas to read structure info for.
+
+    Returns
+    -------
+    list
+        A list of dictionaries containing the structure info
     """
     atlas = BrainGlobeAtlas(atlas_name=atlas_name, check_latest=False)
 

--- a/brainrender_napari/utils/load_user_data.py
+++ b/brainrender_napari/utils/load_user_data.py
@@ -1,26 +1,39 @@
-import json
+from functools import lru_cache
 
-from brainglobe_atlasapi import config
-from brainglobe_atlasapi.list_atlases import get_local_atlas_version
+from brainglobe_atlasapi import BrainGlobeAtlas
 
 
+@lru_cache(maxsize=32)
 def read_atlas_metadata_from_file(atlas_name: str):
-    """Reads atlas metadata stored in a `.json` in the BrainGlobe directory."""
-    brainglobe_dir = config.get_brainglobe_dir()
-    with open(
-        brainglobe_dir
-        / f"{atlas_name}_v{get_local_atlas_version(atlas_name=atlas_name)}"
-        / "metadata.json",
-    ) as metadata_file:
-        return json.loads(metadata_file.read())
+    """Reads atlas metadata from the local manifest in the BrainGlobe
+    directory.
+    """
+    atlas = BrainGlobeAtlas(atlas_name=atlas_name, check_latest=False)
+
+    keys_to_remove = [
+        "coordinate_space",
+        "terminology",
+        "annotation_set",
+        "template",
+    ]
+
+    metadata_dict = atlas.metadata
+    metadata_dict = {
+        k: v for k, v in metadata_dict.items() if k not in keys_to_remove
+    }
+    additional_references = metadata_dict.get("additional_references", [])
+    if additional_references:
+        additional_references = [ref["name"] for ref in additional_references]
+        metadata_dict["additional_references"] = additional_references
+
+    return metadata_dict
 
 
+@lru_cache(maxsize=32)
 def read_atlas_structures_from_file(atlas_name: str):
-    """Reads structure info from a '.json' in the BrainGlobe directory."""
-    brainglobe_dir = config.get_brainglobe_dir()
-    with open(
-        brainglobe_dir
-        / f"{atlas_name}_v{get_local_atlas_version(atlas_name=atlas_name)}"
-        / "structures.json",
-    ) as metadata_file:
-        return json.loads(metadata_file.read())
+    """Reads atlas structure info from the local terminology file in the
+    BrainGlobe directory.
+    """
+    atlas = BrainGlobeAtlas(atlas_name=atlas_name, check_latest=False)
+
+    return atlas.structures_list

--- a/brainrender_napari/utils/load_user_data.py
+++ b/brainrender_napari/utils/load_user_data.py
@@ -3,7 +3,7 @@ from functools import lru_cache
 from brainglobe_atlasapi import BrainGlobeAtlas
 
 
-@lru_cache(maxsize=32)
+@lru_cache()
 def read_atlas_metadata_from_file(atlas_name: str):
     """Reads atlas metadata from the local manifest in the BrainGlobe
     directory. Removes nested keys that are not relevant for the user.
@@ -40,7 +40,7 @@ def read_atlas_metadata_from_file(atlas_name: str):
     return metadata_dict
 
 
-@lru_cache(maxsize=32)
+@lru_cache()
 def read_atlas_structures_from_file(atlas_name: str):
     """Reads atlas structure info from the local terminology file in the
     BrainGlobe directory.

--- a/brainrender_napari/widgets/atlas_manager_view.py
+++ b/brainrender_napari/widgets/atlas_manager_view.py
@@ -11,6 +11,7 @@ that any interested observers can connect to.
 
 from typing import Callable
 
+from brainglobe_atlasapi import BrainGlobeAtlas
 from brainglobe_atlasapi.list_atlases import (
     get_all_atlases_lastversions,
     get_atlases_lastversions,
@@ -30,8 +31,8 @@ from brainrender_napari.widgets.atlas_manager_dialog import AtlasManagerDialog
 
 
 class AtlasManagerView(QTableView):
-    download_atlas_confirmed = Signal(str)
-    update_atlas_confirmed = Signal(str)
+    download_atlas_confirmed = Signal()
+    update_atlas_confirmed = Signal()
     progress_updated = Signal(
         int, int, str, object
     )  # completed, total, atlas_name, operation_type
@@ -87,38 +88,54 @@ class AtlasManagerView(QTableView):
         atlas_name: str = self.selected_atlas_name()
 
         worker = self._apply_in_thread(
-            install_atlas,
+            self._download_atlas_install, atlas_name
+        )
+        worker.returned.connect(
+            lambda: [
+                self.download_atlas_confirmed.emit(),
+                self.source_model.refresh_data(),
+            ]
+        )
+        worker.start()
+
+    def _download_atlas_install(self, atlas_name: str) -> None:
+        install_atlas(
             atlas_name,
             fn_update=lambda completed, total: self.progress_updated.emit(
                 completed, total, atlas_name, "Downloading"
             ),
         )
-        worker.returned.connect(
-            lambda result: [
-                self.download_atlas_confirmed.emit(result),
-                self.source_model.refresh_data(),
-            ]
-        )
-        worker.start()
+        self._download_atlas_images(atlas_name)
+
+    def _download_atlas_images(self, atlas_name: str) -> None:
+        """Downloads the images for the specified atlas."""
+        atlas = BrainGlobeAtlas(atlas_name, check_latest=False)
+        _ = atlas.template
+        _ = atlas.annotation
+        _ = atlas.hemispheres
 
     def _on_update_atlas_confirmed(self) -> None:
         """Updates the currently selected atlas and signals this."""
         atlas_name = self.selected_atlas_name()
 
-        worker = self._apply_in_thread(
-            update_atlas,
+        worker = self._apply_in_thread(self._download_atlas_update, atlas_name)
+        worker.returned.connect(
+            lambda: [
+                self.update_atlas_confirmed.emit(),
+                self.source_model.refresh_data(),
+            ]
+        )
+        worker.start()
+
+    def _download_atlas_update(self, atlas_name: str) -> None:
+        """Updates the specified atlas."""
+        update_atlas(
             atlas_name,
             fn_update=lambda completed, total: self.progress_updated.emit(
                 completed, total, atlas_name, "Updating"
             ),
         )
-        worker.returned.connect(
-            lambda result: [
-                self.update_atlas_confirmed.emit(result),
-                self.source_model.refresh_data(),
-            ]
-        )
-        worker.start()
+        self._download_atlas_images(atlas_name)
 
     def selected_atlas_name(self) -> str:
         """A single place to get a valid selected atlas name."""

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ classifiers = [
 ]
 
 dependencies = [
-    "brainglobe-atlasapi>=2.2.0",
+    "brainglobe-atlasapi>=3.0.0rc1",
     "brainglobe-utils>=0.4.3",
     "meshio",
     "napari>=0.6.1",

--- a/pyproject.toml
+++ b/pyproject.toml
@@ -23,7 +23,7 @@ classifiers = [
 ]
 
 dependencies = [
-    "brainglobe-atlasapi>=3.0.0rc1",
+    "brainglobe-atlasapi>=3.0.0rc2",
     "brainglobe-utils>=0.4.3",
     "meshio",
     "napari>=0.6.1",

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -1,11 +1,36 @@
 import json
 import os
-import shutil
 from pathlib import Path
 
 import pytest
-from brainglobe_atlasapi import BrainGlobeAtlas, config, list_atlases
+from brainglobe_atlasapi import (
+    BrainGlobeAtlas,
+    config,
+    descriptors,
+    list_atlases,
+)
+from brainglobe_atlasapi.list_atlases import (
+    get_downloaded_atlases,
+    get_local_atlas_version,
+)
 from qtpy.QtCore import Qt
+
+
+def _local_atlas_version_dir(atlas_name, version=None):
+    """Local directory holding a specific version of a downloaded atlas.
+
+    Mirrors the layout the atlas API uses on disk. If ``version`` is None the
+    currently downloaded version (folder form, e.g. ``3_0``) is used.
+    """
+    if version is None:
+        version = get_local_atlas_version(atlas_name).replace(".", "_")
+    return (
+        config.get_brainglobe_dir()
+        / "brainglobe-atlasapi"
+        / descriptors.V3_ATLAS_ROOTDIR
+        / atlas_name
+        / version
+    )
 
 
 @pytest.fixture(autouse=True)
@@ -56,26 +81,41 @@ def setup_preexisting_local_atlases():
     """Automatically setup all tests to have three downloaded atlases
     in the test user data."""
     preexisting_atlases = [
-        ("example_mouse_100um", "v1.2"),
-        ("allen_mouse_100um", "v1.2"),
-        ("osten_mouse_100um", "v1.1"),
+        "example_mouse_100um",
+        "allen_mouse_100um",
+        "osten_mouse_100um",
     ]
-    for atlas_name, version in preexisting_atlases:
-        if not Path.exists(
-            Path.home() / f".brainglobe/{atlas_name}_{version}"
-        ):
+    downloaded_atlases = get_downloaded_atlases()
+    for atlas_name in preexisting_atlases:
+        if atlas_name not in downloaded_atlases:
             _ = BrainGlobeAtlas(atlas_name)
 
-    # mock an additional reference for the example mouse
-    atlas_path = Path.home() / ".brainglobe" / "example_mouse_100um_v1.2"
-    metadata_path = atlas_path / "metadata.json"
-    if metadata_path.exists():
-        with open(metadata_path, "r") as f:
-            metadata = f.read()
-            metadata_dict = json.loads(metadata)
-            metadata_dict["additional_references"] = ["reference"]
-            with open(metadata_path, "w") as f:
-                json.dump(metadata_dict, f, indent=4)
+    # Mock an additional reference for the example mouse.
+    manifest_path = (
+        _local_atlas_version_dir("example_mouse_100um") / "manifest.json"
+    )
+    if manifest_path.exists():
+        with open(manifest_path, "r") as f:
+            metadata_dict = json.loads(f.read())
+        metadata_dict["additional_references"] = [
+            {
+                "name": "reference",
+                "location": metadata_dict["template"]["location"],
+            }
+        ]
+        with open(manifest_path, "w") as f:
+            json.dump(metadata_dict, f, indent=4)
+
+
+@pytest.fixture
+def atlas_row():
+    """Look up a model row by atlas name."""
+
+    def _atlas_row(model, atlas_name):
+        names = [model.index(row, 0).data() for row in range(model.rowCount())]
+        return names.index(atlas_name)
+
+    return _atlas_row
 
 
 @pytest.fixture
@@ -104,24 +144,32 @@ def double_click_on_view(qtbot):
 
 @pytest.fixture
 def mock_newer_atlas_version_available():
-    current_version_path = Path.home() / ".brainglobe/example_mouse_100um_v1.2"
-    older_version_path = Path.home() / ".brainglobe/example_mouse_100um_v1.1"
+    """Simulate a locally downloaded atlas being out of date."""
+    assert (
+        list_atlases.folder_version_to_dotted(
+            list_atlases.get_local_atlas_version("example_mouse_100um")
+        )
+        == list_atlases.get_atlases_lastversions()["example_mouse_100um"][
+            "latest_version"
+        ]
+    ), "example_mouse_100um is expected to be up to date before mocking"
+
+    older_version = "2_0"
+    current_version_path = _local_atlas_version_dir("example_mouse_100um")
+    older_version_path = _local_atlas_version_dir(
+        "example_mouse_100um", older_version
+    )
     assert current_version_path.exists() and not older_version_path.exists()
 
     current_version_path.rename(older_version_path)
     assert older_version_path.exists() and not current_version_path.exists()
-    assert (
-        list_atlases.get_atlases_lastversions()["example_mouse_100um"][
-            "latest_version"
-        ]
-        == "1.2"
-    )
-    assert list_atlases.get_local_atlas_version("example_mouse_100um") == "1.1"
+    assert list_atlases.get_local_atlas_version(
+        "example_mouse_100um"
+    ) == older_version.replace("_", ".")
 
     yield  # run test with outdated version
 
     # cleanup: ensure version is up-to-date again
     if older_version_path.exists():
-        shutil.rmtree(path=older_version_path)
-        _ = BrainGlobeAtlas("example_mouse_100um")
+        older_version_path.rename(current_version_path)
     assert current_version_path.exists() and not older_version_path.exists()

--- a/tests/conftest.py
+++ b/tests/conftest.py
@@ -15,6 +15,11 @@ from brainglobe_atlasapi.list_atlases import (
 )
 from qtpy.QtCore import Qt
 
+from brainrender_napari.utils.load_user_data import (
+    read_atlas_metadata_from_file,
+    read_atlas_structures_from_file,
+)
+
 
 def _local_atlas_version_dir(atlas_name, version=None):
     """Local directory holding a specific version of a downloaded atlas.
@@ -105,6 +110,42 @@ def setup_preexisting_local_atlases():
         ]
         with open(manifest_path, "w") as f:
             json.dump(metadata_dict, f, indent=4)
+
+
+@pytest.fixture(autouse=True)
+def clear_atlas_read_caches():
+    """Drop the `lru_cache`s on the atlas readers between tests.
+
+    They are keyed on the atlas name alone, so a value cached before a
+    fixture rearranges the atlas folder would otherwise leak into the next
+    test.
+    """
+    read_atlas_metadata_from_file.cache_clear()
+    read_atlas_structures_from_file.cache_clear()
+    yield
+    read_atlas_metadata_from_file.cache_clear()
+    read_atlas_structures_from_file.cache_clear()
+
+
+@pytest.fixture
+def mock_brainglobe_dir(tmp_path, monkeypatch):
+    """Point the atlas API's default brainglobe directory at an empty
+    `tmp_path`, so tests that fetch real atlas data neither benefit from nor
+    disturb the shared (test) installation."""
+    monkeypatch.setattr(
+        config, "CONFIG_PATH", tmp_path / config.CONFIG_FILENAME
+    )
+    monkeypatch.setattr(
+        config,
+        "TEMPLATE_CONF_DICT",
+        {
+            "default_dirs": {
+                "brainglobe_dir": tmp_path,
+                "interm_download_dir": tmp_path,
+            }
+        },
+    )
+    return tmp_path
 
 
 @pytest.fixture

--- a/tests/test_integration/test_brainrender_viewer_widget.py
+++ b/tests/test_integration/test_brainrender_viewer_widget.py
@@ -61,7 +61,7 @@ def test_double_click_on_locally_available_atlas_row(
     add_atlas_to_viewer_mock.assert_called_once()
 
 
-def test_structure_row_double_clicked(viewer_widget, mocker):
+def test_structure_row_double_clicked(viewer_widget, mocker, atlas_row):
     """Checks that when the structure view widgets emit "VS" and
     the allen_mouse_100um atlas is selected, the NapariAtlasRepresentation
     function is called in the expected way.
@@ -70,15 +70,16 @@ def test_structure_row_double_clicked(viewer_widget, mocker):
         "brainrender_napari.brainrender_viewer_widget"
         ".NapariAtlasRepresentation.add_structure_to_viewer"
     )
-    viewer_widget.atlas_viewer_view.selectRow(
-        4
-    )  # allen_mouse_100um is in row 4
+    row = atlas_row(
+        viewer_widget.atlas_viewer_view.model(), "allen_mouse_100um"
+    )
+    viewer_widget.atlas_viewer_view.selectRow(row)
 
     viewer_widget.structure_view.add_structure_requested.emit("VS")
     add_structure_to_viewer_mock.assert_called_once_with("VS")
 
 
-def test_add_additional_reference_selected(viewer_widget, mocker):
+def test_add_additional_reference_selected(viewer_widget, mocker, atlas_row):
     """Checks that when the atlas viewer view requests an additional
     reference, the NapariAtlasRepresentation function is called in
     the expected way."""
@@ -86,9 +87,11 @@ def test_add_additional_reference_selected(viewer_widget, mocker):
         "brainrender_napari.brainrender_viewer_widget"
         ".NapariAtlasRepresentation.add_additional_reference"
     )
-    viewer_widget.atlas_viewer_view.selectRow(
-        0
-    )  # # example atlas + mock additional reference is in row 0
+    # example atlas has a mock additional reference set up in conftest
+    row = atlas_row(
+        viewer_widget.atlas_viewer_view.model(), "example_mouse_100um"
+    )
+    viewer_widget.atlas_viewer_view.selectRow(row)
     assert (
         viewer_widget.atlas_viewer_view.selected_atlas_name()
         == "example_mouse_100um"
@@ -102,13 +105,14 @@ def test_add_additional_reference_selected(viewer_widget, mocker):
     )
 
 
-def test_show_structures_checkbox(viewer_widget, mocker):
+def test_show_structures_checkbox(viewer_widget, mocker, atlas_row):
     structure_view_refresh_mock = mocker.patch(
         "brainrender_napari.brainrender_viewer_widget.StructureView.refresh"
     )
-    viewer_widget.atlas_viewer_view.selectRow(
-        0
-    )  # example_mouse_100um is in row 0
+    row = atlas_row(
+        viewer_widget.atlas_viewer_view.model(), "example_mouse_100um"
+    )
+    viewer_widget.atlas_viewer_view.selectRow(row)
     structure_view_refresh_mock.assert_called_with(
         "example_mouse_100um", False
     )

--- a/tests/test_unit/test_atlas_manager_view.py
+++ b/tests/test_unit/test_atlas_manager_view.py
@@ -10,6 +10,17 @@ def atlas_manager_view(qtbot):
     return AtlasManagerView()
 
 
+@pytest.fixture(autouse=True)
+def mock_download_atlas_images(mocker):
+    """Don't pull image data for real after a (mocked) download/update.
+
+    Downloading/updating is always mocked here, so the atlas the prefetch
+    step is given may not exist at all; letting it run raises inside the
+    thread worker and leaks into the next test's Qt event loop.
+    """
+    return mocker.patch.object(AtlasManagerView, "_download_atlas_images")
+
+
 def test_update_atlas_confirmed(
     qtbot,
     mocker,
@@ -27,26 +38,26 @@ def test_update_atlas_confirmed(
         return_value="example_mouse_100um",
     )
 
-    mocker.patch(
+    update_atlas_mock = mocker.patch(
         "brainrender_napari.widgets.atlas_manager_view.update_atlas",
-        return_value="example_mouse_100um",
     )
 
     with qtbot.waitSignal(
         atlas_manager_view.update_atlas_confirmed,
         timeout=300000,  # assumes atlas can be updated in 5 minutes!
-    ) as update_atlas_confirmed_signal:
+    ):
         atlas_manager_view._on_update_atlas_confirmed()
 
-    assert update_atlas_confirmed_signal.args == ["example_mouse_100um"]
+    assert update_atlas_mock.call_args.args == ("example_mouse_100um",)
 
 
 def test_progress_signal_emission(atlas_manager_view, qtbot, mocker):
     """Test that progress_updated signal is emitted with correct parameters."""
     mocker.patch(
         "brainrender_napari.widgets.atlas_manager_view.update_atlas",
-        side_effect=lambda atlas_name, fn_update: fn_update(50, 100)
-        or atlas_name,
+        side_effect=lambda atlas_name, fn_update: (
+            fn_update(50, 100) or atlas_name
+        ),
     )
 
     mocker.patch.object(
@@ -88,18 +99,14 @@ def test_double_click_on_not_yet_downloaded_atlas_row(
     dialog_exec_mock.assert_called_once()
 
 
-def test_download_confirmed_callback(atlas_manager_view, qtbot, mocker):
-    """Checks that confirming atlas download creates local copy of
-    example atlas files and emits expected signal.
-
-    Test setup consists of remembering the expected files and folders
-    of a preexisting atlas and then removing them. This allows checking
-    that the function triggers the creation of the same local copy
-    of the atlas as the `brainglobe_atlasapi` itself.
+def test_download_confirmed_callback(
+    atlas_manager_view, qtbot, mocker, mock_download_atlas_images
+):
+    """Checks that confirming atlas download installs the atlas, pulls its
+    image data and emits the expected signal.
     """
-    mocker.patch(
+    install_atlas_mock = mocker.patch(
         "brainrender_napari.widgets.atlas_manager_view.install_atlas",
-        return_value="example_mouse_100um",
     )
 
     # Mock selected_atlas_name to ensure consistent return value
@@ -112,11 +119,12 @@ def test_download_confirmed_callback(atlas_manager_view, qtbot, mocker):
     with qtbot.waitSignal(
         atlas_manager_view.download_atlas_confirmed,
         timeout=300000,  # assumes atlas can be installed in 5 minutes!
-    ) as download_atlas_confirmed_signal:
+    ):
         # Directly call the download confirmation method
         atlas_manager_view._on_download_atlas_confirmed()
 
-    assert download_atlas_confirmed_signal.args == ["example_mouse_100um"]
+    assert install_atlas_mock.call_args.args == ("example_mouse_100um",)
+    mock_download_atlas_images.assert_called_once_with("example_mouse_100um")
 
 
 def test_double_click_on_outdated_atlas_row(

--- a/tests/test_unit/test_atlas_manager_view.py
+++ b/tests/test_unit/test_atlas_manager_view.py
@@ -1,3 +1,5 @@
+import json
+
 import pytest
 from brainglobe_atlasapi import BrainGlobeAtlas
 from brainglobe_atlasapi.descriptors import (
@@ -280,6 +282,58 @@ def test_apply_in_thread(qtbot, mocker):
 
     # Restore the original _apply_in_thread
     atlas_manager_view._apply_in_thread = original_apply_in_thread
+
+
+def test_download_atlas_install(qtbot, mock_brainglobe_dir):
+    """Check that downloading an atlas installs it in the correct location."""
+    atlas_manager_view = AtlasManagerView()
+
+    atlas_path = (
+        mock_brainglobe_dir
+        / "brainglobe-atlasapi"
+        / "atlases"
+        / "example_mouse_100um"
+    )
+    assert not atlas_path.exists(), "Atlas should not exist before download"
+
+    atlas_manager_view._download_atlas_install("example_mouse_100um")
+
+    assert atlas_path.exists(), "Atlas should exist after download"
+
+
+def test_download_atlas_update(qtbot, mock_brainglobe_dir):
+    """Check that updating an atlas updates it in the correct location."""
+    atlas_manager_view = AtlasManagerView()
+
+    atlas = BrainGlobeAtlas("example_mouse_100um")
+
+    # Simulate an older version of the atlas
+    current_version = atlas.local_version
+    current_version_str = "_".join([str(ver) for ver in atlas.local_version])
+    old_version = "1_0"
+    atlas_dir = atlas.brainglobe_dir / atlas.metadata["location"][1:]
+
+    atlas.metadata["version"] = old_version.replace("_", ".")
+
+    # Overwrite the manifest.json to reflect the old version
+    with open(atlas_dir / "manifest.json", "w") as f:
+        json.dump(atlas.metadata, f, indent=4)
+
+    atlas_dir.rename(atlas_dir.parent / old_version)
+
+    assert not (
+        atlas_dir.parent / current_version_str
+    ).exists(), "current version should not exist before update"
+
+    atlas_manager_view._download_atlas_update("example_mouse_100um")
+
+    # Check that the version has been updated
+    assert (
+        atlas_dir.parent / current_version_str
+    ).exists(), "current version should exist after update"
+
+    atlas = BrainGlobeAtlas("example_mouse_100um")
+    assert atlas.local_version == current_version
 
 
 def test_download_atlas_images(mock_brainglobe_dir, qtbot):

--- a/tests/test_unit/test_atlas_manager_view.py
+++ b/tests/test_unit/test_atlas_manager_view.py
@@ -1,4 +1,9 @@
 import pytest
+from brainglobe_atlasapi import BrainGlobeAtlas
+from brainglobe_atlasapi.descriptors import (
+    V3_ANNOTATION_NAME,
+    V3_TEMPLATE_NAME,
+)
 from qtpy.QtCore import Qt
 
 from brainrender_napari.utils.formatting import format_atlas_name
@@ -10,7 +15,7 @@ def atlas_manager_view(qtbot):
     return AtlasManagerView()
 
 
-@pytest.fixture(autouse=True)
+@pytest.fixture
 def mock_download_atlas_images(mocker):
     """Don't pull image data for real after a (mocked) download/update.
 
@@ -26,6 +31,7 @@ def test_update_atlas_confirmed(
     mocker,
     atlas_manager_view,
     mock_newer_atlas_version_available,
+    mock_download_atlas_images,
 ):
     """
     Test that confirming atlas update triggers the correct actions
@@ -49,9 +55,12 @@ def test_update_atlas_confirmed(
         atlas_manager_view._on_update_atlas_confirmed()
 
     assert update_atlas_mock.call_args.args == ("example_mouse_100um",)
+    mock_download_atlas_images.assert_called_once_with("example_mouse_100um")
 
 
-def test_progress_signal_emission(atlas_manager_view, qtbot, mocker):
+def test_progress_signal_emission(
+    atlas_manager_view, qtbot, mocker, mock_download_atlas_images
+):
     """Test that progress_updated signal is emitted with correct parameters."""
     mocker.patch(
         "brainrender_napari.widgets.atlas_manager_view.update_atlas",
@@ -271,3 +280,32 @@ def test_apply_in_thread(qtbot, mocker):
 
     # Restore the original _apply_in_thread
     atlas_manager_view._apply_in_thread = original_apply_in_thread
+
+
+def test_download_atlas_images(mock_brainglobe_dir, qtbot):
+    """Downloading an atlas only fetches its metadata; the image chunks are
+    pulled lazily on first access. Check `_download_atlas_images` actually
+    pulls them, so the atlas is usable offline afterwards.
+    """
+    atlas_manager_view = AtlasManagerView()
+
+    atlas = BrainGlobeAtlas("example_mouse_100um", check_latest=False)
+    template_zarr = (
+        atlas.root_dir
+        / atlas.metadata["annotation_set"]["template"]["location"][1:]
+        / V3_TEMPLATE_NAME
+    )
+    annotation_zarr = (
+        atlas.root_dir
+        / atlas.metadata["annotation_set"]["location"][1:]
+        / V3_ANNOTATION_NAME
+    )
+
+    # chunks live in a `c` directory under the resolution level in use
+    assert not any(template_zarr.glob("*/c")), "template chunks pre-exist"
+    assert not any(annotation_zarr.glob("*/c")), "annotation chunks pre-exist"
+
+    atlas_manager_view._download_atlas_images("example_mouse_100um")
+
+    assert any(template_zarr.glob("*/c")), "template chunks not downloaded"
+    assert any(annotation_zarr.glob("*/c")), "annotation chunks not downloaded"

--- a/tests/test_unit/test_atlas_table_model.py
+++ b/tests/test_unit/test_atlas_table_model.py
@@ -60,15 +60,20 @@ def test_model_header_invalid_view():
     ],
 )
 def test_background_color_for_outdated_atlas(
-    atlas_table_model, theme, expected_rgb, mock_newer_atlas_version_available
+    atlas_table_model,
+    theme,
+    expected_rgb,
+    mock_newer_atlas_version_available,
+    atlas_row,
 ):
     """
-    For out-of-date atlas (local_version=“1.1”, latest_version=“1.2”),
-    Test to verify that the amber color is returned according to the theme.
+    For an out-of-date atlas (local version older than the latest version),
+    test to verify that the amber color is returned according to the theme.
     """
     get_settings().appearance.theme = theme
 
-    index = atlas_table_model.index(0, 0)
+    row = atlas_row(atlas_table_model, "example_mouse_100um")
+    index = atlas_table_model.index(row, 0)
     brush = atlas_table_model.data(index, role=Qt.BackgroundRole)
     assert isinstance(brush, QBrush)
 
@@ -78,3 +83,17 @@ def test_background_color_for_outdated_atlas(
     assert (
         actual_rgb == expected_rgb
     ), f"Expected RGB {expected_rgb}, but got {actual_rgb}"
+
+
+def test_no_background_color_for_up_to_date_atlas(mocker, atlas_row):
+    """A downloaded atlas that is up to date should have no special
+    background. This guards against the local version (folder form, e.g.
+    "3_0") failing to compare equal to the latest version (dotted, "3.0").
+    """
+    mock_view = mocker.Mock(spec=["get_tooltip_text"])
+    model = AtlasTableModel(view_type=mock_view)
+
+    # allen_mouse_100um is downloaded and at the latest version
+    row = atlas_row(model, "allen_mouse_100um")
+    index = model.index(row, 0)
+    assert model.data(index, role=Qt.BackgroundRole) is None

--- a/tests/test_unit/test_atlas_table_model.py
+++ b/tests/test_unit/test_atlas_table_model.py
@@ -4,6 +4,10 @@ from qtpy.QtCore import Qt
 from qtpy.QtGui import QBrush
 
 from brainrender_napari.data_models.atlas_table_model import AtlasTableModel
+from brainrender_napari.utils.load_user_data import (
+    read_atlas_metadata_from_file,
+    read_atlas_structures_from_file,
+)
 
 
 @pytest.fixture
@@ -97,3 +101,19 @@ def test_no_background_color_for_up_to_date_atlas(mocker, atlas_row):
     row = atlas_row(model, "allen_mouse_100um")
     index = model.index(row, 0)
     assert model.data(index, role=Qt.BackgroundRole) is None
+
+
+def test_refresh_data_clears_atlas_read_caches(mocker):
+    """A refresh follows a download or update, so the cached metadata and
+    structure reads must not survive it."""
+    model = AtlasTableModel(view_type=mocker.Mock(spec=["get_tooltip_text"]))
+
+    read_atlas_metadata_from_file("example_mouse_100um")
+    read_atlas_structures_from_file("example_mouse_100um")
+    assert read_atlas_metadata_from_file.cache_info().currsize == 1
+    assert read_atlas_structures_from_file.cache_info().currsize == 1
+
+    model.refresh_data()
+
+    assert read_atlas_metadata_from_file.cache_info().currsize == 0
+    assert read_atlas_structures_from_file.cache_info().currsize == 0

--- a/tests/test_unit/test_atlas_viewer_view.py
+++ b/tests/test_unit/test_atlas_viewer_view.py
@@ -19,16 +19,17 @@ def atlas_viewer_view(qtbot) -> AtlasViewerView:
 
 
 @pytest.mark.parametrize(
-    "row, expected_atlas_name",
+    "expected_atlas_name",
     [
-        (0, "example_mouse_100um"),
-        (4, "allen_mouse_100um"),
+        "example_mouse_100um",
+        "allen_mouse_100um",
     ],
 )
 def test_atlas_view_valid_selection(
-    row, expected_atlas_name, atlas_viewer_view
+    expected_atlas_name, atlas_viewer_view, atlas_row
 ):
     """Checks selected_atlas_name for valid current indices"""
+    row = atlas_row(atlas_viewer_view.model(), expected_atlas_name)
     model_index = atlas_viewer_view.model().index(row, 0)
     atlas_viewer_view.setCurrentIndex(model_index)
     assert atlas_viewer_view.selected_atlas_name() == expected_atlas_name
@@ -42,14 +43,16 @@ def test_atlas_view_invalid_selection(atlas_viewer_view):
         atlas_viewer_view.selected_atlas_name()
 
 
-def test_atlas_view_not_downloaded_selection(qtbot, atlas_viewer_view):
+def test_atlas_view_not_downloaded_selection(
+    qtbot, atlas_viewer_view, atlas_row
+):
     """Checks that selected_atlas_name raises an assertion error
     if current index is valid, but not a downloaded atlas.
     """
+    # human atlas is not part of the downloaded test data
+    row = atlas_row(atlas_viewer_view.model(), "allen_human_500um")
     with qtbot.capture_exceptions() as exceptions:
-        # should raise because human atlas (row 6) is not available
-        # exception raised within qt loop in this case.
-        model_index = atlas_viewer_view.model().index(6, 0)
+        model_index = atlas_viewer_view.model().index(row, 0)
         atlas_viewer_view.setCurrentIndex(model_index)
     assert len(exceptions) == 1
     _, exception, collected_traceback = exceptions[0]  # ignore type
@@ -72,19 +75,24 @@ def test_hover_atlas_viewer_view(atlas_viewer_view, mocker):
 
 
 @pytest.mark.parametrize(
-    "row,expected_atlas_name",
+    "expected_atlas_name",
     [
-        (0, "example_mouse_100um"),
-        (4, "allen_mouse_100um"),
-        (14, "osten_mouse_100um"),
+        "example_mouse_100um",
+        "allen_mouse_100um",
+        "osten_mouse_100um",
     ],
 )
 def test_double_click_on_locally_available_atlas_row(
-    atlas_viewer_view, double_click_on_view, qtbot, row, expected_atlas_name
+    atlas_viewer_view,
+    double_click_on_view,
+    qtbot,
+    expected_atlas_name,
+    atlas_row,
 ):
     """Check for a few locally available low-res atlases that double-clicking
     them on the atlas table view emits a signal with their expected names.
     """
+    row = atlas_row(atlas_viewer_view.model(), expected_atlas_name)
     model_index = atlas_viewer_view.model().index(row, 1)
     atlas_viewer_view.setCurrentIndex(model_index)
 
@@ -96,16 +104,18 @@ def test_double_click_on_locally_available_atlas_row(
     assert add_atlas_requested_signal.args == [expected_atlas_name]
 
 
-def test_additional_reference_menu(atlas_viewer_view, qtbot, mocker):
+def test_additional_reference_menu(
+    atlas_viewer_view, qtbot, mocker, atlas_row
+):
     """Checks callback to additional reference menu calls QMenu exec
     and emits expected signal"""
-    atlas_viewer_view.selectRow(
-        0
-    )  # example atlas + mock additional reference is in row 0
+    # example atlas has a mock additional reference set up in conftest
+    row = atlas_row(atlas_viewer_view.model(), "example_mouse_100um")
+    atlas_viewer_view.selectRow(row)
     from qtpy.QtCore import QPoint
     from qtpy.QtWidgets import QAction
 
-    x = atlas_viewer_view.rowViewportPosition(0)
+    x = atlas_viewer_view.rowViewportPosition(row)
     y = atlas_viewer_view.columnViewportPosition(1)
     position = QPoint(x, y)
     qmenu_exec_mock = mocker.patch(

--- a/tests/test_unit/test_metadata_reading.py
+++ b/tests/test_unit/test_metadata_reading.py
@@ -6,8 +6,42 @@ from brainrender_napari.utils.load_user_data import (
 
 
 def test_metadata_reading():
-    """Checks that metadata read from file matches original metadata"""
+    """Checks that metadata read via the atlas API is the atlas' own
+    metadata, minus the nested entries that aren't useful in a tooltip.
+    """
     atlas = BrainGlobeAtlas("example_mouse_100um")
-    expected_metadata = atlas.metadata
     file_metadata = read_atlas_metadata_from_file(atlas.atlas_name)
-    assert file_metadata == expected_metadata
+
+    dropped_keys = {
+        "coordinate_space",
+        "terminology",
+        "annotation_set",
+        "template",
+    }
+    assert dropped_keys.isdisjoint(file_metadata)
+    expected = {
+        key: value
+        for key, value in atlas.metadata.items()
+        if key not in dropped_keys and key != "additional_references"
+    }
+    assert {
+        key: value
+        for key, value in file_metadata.items()
+        if key != "additional_references"
+    } == expected
+
+
+def test_additional_references_are_flattened_to_names():
+    """The additional references are dicts in the atlas metadata, but only
+    their names are shown in the UI. `conftest` mocks one onto the example
+    atlas.
+    """
+    atlas = BrainGlobeAtlas("example_mouse_100um")
+    expected_names = [
+        reference["name"]
+        for reference in atlas.metadata["additional_references"]
+    ]
+    assert expected_names, "test setup should mock an additional reference"
+
+    file_metadata = read_atlas_metadata_from_file(atlas.atlas_name)
+    assert file_metadata["additional_references"] == expected_names

--- a/tests/test_unit/test_metadata_reading.py
+++ b/tests/test_unit/test_metadata_reading.py
@@ -1,7 +1,9 @@
+import pytest
 from brainglobe_atlasapi import BrainGlobeAtlas
 
 from brainrender_napari.utils.load_user_data import (
     read_atlas_metadata_from_file,
+    read_atlas_structures_from_file,
 )
 
 
@@ -45,3 +47,33 @@ def test_additional_references_are_flattened_to_names():
 
     file_metadata = read_atlas_metadata_from_file(atlas.atlas_name)
     assert file_metadata["additional_references"] == expected_names
+
+
+def test_structures_reading():
+    """Structures are read via the atlas API and passed through as-is."""
+    atlas = BrainGlobeAtlas("example_mouse_100um")
+    assert (
+        read_atlas_structures_from_file(atlas.atlas_name)
+        == atlas.structures_list
+    )
+
+
+@pytest.mark.parametrize(
+    "read_from_file",
+    [read_atlas_metadata_from_file, read_atlas_structures_from_file],
+)
+def test_reads_are_cached(mocker, read_from_file):
+    """Both readers instantiate a `BrainGlobeAtlas`, which is expensive, so
+    repeated reads of the same atlas should be served from the cache."""
+    atlas_mock = mocker.patch(
+        "brainrender_napari.utils.load_user_data.BrainGlobeAtlas"
+    )
+    atlas_mock.return_value.metadata = {"name": "example_mouse_100um"}
+    atlas_mock.return_value.structures_list = []
+
+    read_from_file("example_mouse_100um")
+    read_from_file("example_mouse_100um")
+
+    atlas_mock.assert_called_once_with(
+        atlas_name="example_mouse_100um", check_latest=False
+    )

--- a/tests/test_unit/test_napari_atlas_representation.py
+++ b/tests/test_unit/test_napari_atlas_representation.py
@@ -35,8 +35,8 @@ def test_add_to_viewer(make_napari_viewer, expected_atlas_name, anisotropic):
         atlas._annotation = atlas.annotation[
             0 : len(atlas.annotation) : 3, 0 : len(atlas.annotation[0]) : 2, :
         ]
-        atlas._reference = atlas.reference[
-            0 : len(atlas.reference) : 3, 0 : len(atlas.reference[0]) : 2, :
+        atlas._template = atlas.template[
+            0 : len(atlas.template) : 3, 0 : len(atlas.template[0]) : 2, :
         ]
 
     atlas_representation = NapariAtlasRepresentation(atlas, viewer)
@@ -194,7 +194,7 @@ def test_viewer_tooltip(
     annotation = viewer.layers[1]
 
     event = QMouseEvent(
-        QEvent.Type.MouseMove,
+        QEvent.MouseMove,
         QPointF(0, 0),  # any pos will do to check text
         Qt.MouseButton.NoButton,
         Qt.MouseButton.NoButton,
@@ -220,7 +220,7 @@ def test_too_quick_mouse_move_keyerror(make_napari_viewer, mocker):
     annotation = viewer.layers[1]
 
     event = QMouseEvent(
-        QEvent.Type.MouseMove,
+        QEvent.MouseMove,
         QPointF(0, 0),  # any pos will do to check text
         Qt.MouseButton.NoButton,
         Qt.MouseButton.NoButton,

